### PR TITLE
Factor Source name updates

### PR DESCRIPTION
**Pull request by danvleju-rdx**
- renamed `Passphrase` to `Password`
- added `PasswordFactorSourceHint`
- added `label` field in `DeviceFactorSourceHint` and renamed `name` to `device_name`

### CODE REVIEW

File: apple/Sources/Sargon/Extensions/Methods/Profile/Factor/PasswordFactorSource+Functions.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//
+//  File.swift
+//
+//
+//  Created by Alexander Cyon on 2024-06-02.
```
Comment (CyonAlexRDX) [inline apple/Sources/Sargon/Extensions/Methods/Profile/Factor/PasswordFactorSource+Functions.swift:5]: Created by Dan on 2024-11-29 :)

Comment (CyonAlexRDX) [inline apple/Sources/Sargon/Extensions/Methods/Profile/Factor/PasswordFactorSource+Functions.swift:5]: Created by Dan on 2024-11-29 :)

Comment (CyonAlexRDX) [reply apple/Sources/Sargon/Extensions/Methods/Profile/Factor/PasswordFactorSource+Functions.swift:5]: we should remove those headers with swiftformat :) and we should format swift code like @matiasbzurovski added in som ancient PR.

in a separate PR..

Comment (danvleju-rdx) [reply apple/Sources/Sargon/Extensions/Methods/Profile/Factor/PasswordFactorSource+Functions.swift:5]: Oh, it was just a rename, but I'll remove this anyway :)

File: crates/sargon-uniffi/src/profile/mfa/mfa_factor_sources/password_factor_source/password_factor_source_hint.rs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+use crate::prelude::*;
+use sargon::PasswordFactorSourceHint as InternalPasswordFactorSourceHint;
+
+/// Properties describing a PasswordFactorSource to help user disambiguate between
+/// it and another one.
+#[derive(Clone, PartialEq, Eq, Hash, InternalConversion, uniffi::Record)]
+pub struct PasswordFactorSourceHint {
+    pub label: String,
```
Comment (CyonAlexRDX) [inline crates/sargon-uniffi/src/profile/mfa/mfa_factor_sources/password_factor_source/password_factor_source_hint.rs:8]: maybe doc that this is manually inputted by user? :)

File: crates/sargon/src/profile/v100/factors/factor_sources/device_factor_source/device_factor_source_hint.rs [hunk 3]
```diff
@@ -13,10 +13,22 @@ use crate::prelude::*;
...
-#[display("{name} {model}")]
+#[display("{label} {device_name} {model}")]
 pub struct DeviceFactorSourceHint {
-    /// "iPhone RED"
-    pub name: String,
+    /// A user-assigned name for the device, intended to help users
+    /// differentiate between multiple devices.
+    ///
+    /// Example: "My Phone"
+    #[serde(default = "label_default")]
```
Comment (CyonAlexRDX) [inline crates/sargon/src/profile/v100/factors/factor_sources/device_factor_source/device_factor_source_hint.rs:22]: 👍

File: crates/sargon/src/profile/v100/factors/factor_sources/ledger_hardware_wallet_factor_source/ledger_hardware_wallet_factor_source.rs [hunk 4]
```diff
@@ -98,6 +98,10 @@ impl BaseIsFactorSource for LedgerHardwareWalletFactorSource {
     fn set_common_properties(&mut self, updated: FactorSourceCommon) {
         self.common = updated
     }
+
+    fn name(&self) -> String {
+        self.hint.name.clone()
```
Comment (CyonAlexRDX) [inline crates/sargon/src/profile/v100/factors/factor_sources/ledger_hardware_wallet_factor_source/ledger_hardware_wallet_factor_source.rs:103]: 👍

File: apple/Sources/Sargon/Extensions/Methods/Profile/Factor/PasswordFactorSource+Functions.swift (hunk 1 above)
Comment (danvleju-rdx) [reply apple/Sources/Sargon/Extensions/Methods/Profile/Factor/PasswordFactorSource+Functions.swift:5]: Was that PR merged though?

File: crates/sargon/src/profile/mfa/mfa_factor_sources/security_questions_factor_source/security_questions_factor_source.rs [hunk 5]
```diff
@@ -279,6 +279,10 @@ impl BaseIsFactorSource
     fn set_common_properties(&mut self, updated: FactorSourceCommon) {
         self.common = updated
     }
+
+    fn name(&self) -> String {
+        "NOT IMPLEMENTED".to_owned()
```
Comment (CyonAlexRDX) [inline crates/sargon/src/profile/mfa/mfa_factor_sources/security_questions_factor_source/security_questions_factor_source.rs:284]: Maybe we should use this instead:

```rust
    fn name(&self) -> String {
        let ids = self
            .sealed_mnemonic
            .security_questions
            .items()
            .into_iter()
            .map(|q| q.id())
            .map(|id| format!("#{:?}", id))
            .join(", ");
        format!("Questions: {}", ids)
    }
```

It ensures if we ever were to forget about this, we at least have something which is somewhat decent.

File: apple/Sources/Sargon/Extensions/Methods/Profile/Factor/PasswordFactorSource+Functions.swift (hunk 1 above)
Comment (matiasbzurovski) [reply apple/Sources/Sargon/Extensions/Methods/Profile/Factor/PasswordFactorSource+Functions.swift:5]: I can take care of this now

File: crates/sargon/src/profile/mfa/mfa_factor_sources/arculus_card_factor_source/arculus_card_factor_source.rs [hunk 6]
```diff
@@ -108,6 +108,10 @@ impl BaseIsFactorSource for ArculusCardFactorSource {
     fn set_common_properties(&mut self, updated: FactorSourceCommon) {
         self.common = updated
     }
+
+    fn name(&self) -> String {
+        self.hint.name.clone()
```
Comment (GhenadieVP) [inline crates/sargon/src/profile/mfa/mfa_factor_sources/arculus_card_factor_source/arculus_card_factor_source.rs:113]: not label? I would use the same terminology for all the hints - `label`.

File: crates/sargon/src/profile/mfa/mfa_factor_sources/off_device_mnemonic_factor_source/off_device_mnemonic_factor_source.rs [hunk 7]
```diff
@@ -108,6 +108,10 @@ impl BaseIsFactorSource for OffDeviceMnemonicFactorSource {
     fn set_common_properties(&mut self, updated: FactorSourceCommon) {
         self.common = updated
     }
+
+    fn name(&self) -> String {
+        self.hint.display_name.value.clone()
```
Comment (GhenadieVP) [inline crates/sargon/src/profile/mfa/mfa_factor_sources/off_device_mnemonic_factor_source/off_device_mnemonic_factor_source.rs:113]: `label`?

File: crates/sargon/src/profile/mfa/mfa_factor_sources/password_factor_source/password_factor_source_hint.rs [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    Serialize,
+    Deserialize,
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    derive_more::Display,
+)]
+#[serde(rename_all = "camelCase")]
```
Comment (GhenadieVP) [inline crates/sargon/src/profile/mfa/mfa_factor_sources/password_factor_source/password_factor_source_hint.rs:15]: Needed?

File: crates/sargon/src/profile/mfa/mfa_factor_sources/security_questions_factor_source/security_questions_factor_source.rs (hunk 5 above)
Comment (GhenadieVP) [reply crates/sargon/src/profile/mfa/mfa_factor_sources/security_questions_factor_source/security_questions_factor_source.rs:284]: Why not have a hint the same way we do for other factor sources?

Comment (danvleju-rdx) [reply crates/sargon/src/profile/mfa/mfa_factor_sources/security_questions_factor_source/security_questions_factor_source.rs]: We might, but let’s postpone this until we implement security questions and use this default name for now.

**Review by CyonAlexRDX — APPROVED** [review]
LGTM, good job!

**Review by GhenadieVP — APPROVED** [review]

**Pull request merged**